pkg/ibmvpc/sg_analysis.go

```diff
@@ -16,18 +16,19 @@ import (
 
 	"github.com/np-guard/models/pkg/connection"
 	"github.com/np-guard/models/pkg/ipblock"
+	"github.com/np-guard/vpc-network-config-analyzer/pkg/common"
 )
 
 type SGAnalyzer struct {
 	sgResource   *vpc1.SecurityGroup
 	ingressRules []*SGRule
 	egressRules  []*SGRule
 	// rules are the default ones; that is, no rules were specified manually
-	isDefault           bool
-	ingressConnectivity *ConnectivityResult
-	egressConnectivity  *ConnectivityResult
-	sgMap               map[string]*SecurityGroup
-	referencedIPblocks  []*ipblock.IPBlock
+	isDefault              bool
+	ingressConnectivityMap map[*ipblock.IPBlock]*ConnectivityResult
+	egressConnectivityMap  map[*ipblock.IPBlock]*ConnectivityResult
```

Comment:
consider a separate type for ` map[*ipblock.IPBlock]*ConnectivityResult`  (so it can be more documented what this represents, and  can added with methods, such as `equal` for tests)

Reply:
could you simplify `ingressConnectivityMap` by only keeping the relevant entries that intersect with actual members of the SG ?  (no need to compute potential `ConnectivityResult` for members that do not exist in such range...)

Reply:
> could you simplify `ingressConnectivityMap` by only keeping the relevant entries that intersect with actual members of the SG ? (no need to compute potential `ConnectivityResult` for members that do not exist in such range...)

do not have access to the members

Reply:
probably can be added from the parser